generate/src/contract/events.rs

```diff
@@ -1,21 +1,149 @@
-use crate::contract::{types, Context};
+use crate::contract::{methods, types, Context};
 use crate::util;
 use anyhow::Result;
 use ethcontract_common::abi::{Event, Hash};
+use ethcontract_common::abiext::EventExt;
 use inflector::Inflector;
 use proc_macro2::{Literal, TokenStream};
 use quote::quote;
 
 pub(crate) fn expand(cx: &Context) -> Result<TokenStream> {
+    let structs_mod = expand_structs_mod(cx)?;
     let filters = expand_filters(cx)?;
     let all_events = expand_all_events(cx);
 
     Ok(quote! {
+        #structs_mod
         #filters
         #all_events
     })
 }
 
+fn expand_structs_mod(cx: &Context) -> Result<TokenStream> {
+    let structs = cx
+        .artifact
+        .abi
+        .events()
+        .map(|event| expand_struct(event))
+        .collect::<Result<Vec<_>>>()?;
+    if structs.is_empty() {
+        return Ok(quote! {});
```

Comment:
If I understand this correctly we only create the `pub mod events` if it is not empty. Why is that better than having the empty module? Not saying it is wrong, just curious.

Reply:
Um, I wouldn't say its better, just an arbitrary decision I made. Let me know if you guys would prefer an empty `events` module.

Reply:
The only reason I can think of is that someone might always do something like `use contract::events::*` (maybe this also happens in a macro). Then this would fail if the module didn't exist but it would be fine if it was empty.
On the other hand it could be confusing that there is an empty module, too.
I am fine with either.

Reply:
I'm not sure either. Maybe @fleupold has an opinion. One advantage of not generating `events` module is that if something assumes its there but it isn't then you get a compiler error. I usually prefer to err on the side of caution.

Reply:
No strong feelings either way (let's be consistent how we handle this for other things such as methods). 
I'd argue you would get a compile error anyways when trying to access anything inside the empty module and just using a module without accessing it is not really useful (and will probably be a clippy warning to start with)

Reply:
RIght now we omit things that would be empty, so I will keep as is then.